test/test_autograd.py

```diff
@@ -4904,6 +4904,16 @@ def test_integer_outputs(self):
         self.assertFalse(out.dtype.is_floating_point)
         self.assertFalse(out.requires_grad)
 
+    def test_linalg_qr_r(self):
```

Comment:
So what's the status of backward with this PR vs PyTorch's existing torch.qr?

Reply:
Add a comment describing why this special autograd function is necessary

Reply:
> So what's the status of backward with this PR vs PyTorch's existing torch.qr?

backwards just works, unless you use `torch.linalg.qr(mode='r')`, in which case it cannot.

> Add a comment describing why this special autograd function is necessary

I'm not sure which function you are talking about. Do you mean to add a comment to explain why the test is necessary?￼

Reply:
Yes, sorry, I was unclear. Please add a comment describing why the test is necessary.

Reply:
ok, fixed by 21375b2855